exec/java-exec/src/main/java/org/apache/drill/exec/planner/physical/ConvertCountToDirectScan.java

```diff
@@ -117,6 +117,24 @@ public void onMatch(RelOptRuleCall call) {
       } else if (aggCall.getArgList().size() == 1) {
       // count(columnName) ==> Agg ( Scan )) ==> columnValueCount
         int index = aggCall.getArgList().get(0);
+
+        if (proj != null) {
+          // project in the middle of Agg and Scan : Only when input of AggCall is a RexInputRef in Project, we find the index of Scan's field.
+          // For instance,
+          // Agg - count($0)
+          //  \
+          //  Proj - Exp={$1}
+          //    \
+          //   Scan (col1, col2).
+          // return count of "col2" in Scan's metadata, if found.
+
+          if (proj.getProjects().get(index) instanceof RexInputRef) {
+            index = ((RexInputRef) proj.getProjects().get(index)).getIndex();
```

Comment:
Calcite rewrote count(100) or count(1) into count() ==> aggCall.getArgList.isEmpty() is true. So Line 113 will take care of those cases.